packages/api-client/index.ts

```diff
@@ -0,0 +1,215 @@
+        body
+      ),
+    delete: async (id: string | number, jwt: string): Promise<void> =>
+      this.req("DELETE", `/plans/${id}`, undefined, {
+        Authorization: `Bearer ${jwt}`,
+      }),
+  };
+}
+
+interface SearchClass {
+  name: string;
+  classId: string;
+  subject: string;
+  prereqs?: INEUAndPrereq | INEUOrPrereq;
+  coreqs?: INEUAndPrereq | INEUOrPrereq;
+  maxCredits: number;
+  minCredits: number;
+}
+
+/**
+ * A client for interacting with the Search API. Allows us to fetch
+ * and search for courses.
+ */
+class SearchAPIClient {
+  private axios: AxiosInstance;
+
+  constructor(baseURL = "https://api.searchneu.com/graphql") {
+    this.axios = Axios.create({ baseURL: baseURL });
+  }
+
+  fetchCourse = async (
+    subject: string,
+    classId: string
+  ): Promise<ScheduleCourse> => {
+    const res = await this.axios({
+      method: "post",
+      data: {
+        query: `{ 
+          class(subject: "${subject}", classId: "${classId}") {
+            latestOccurrence {
+              name
+              subject
+              classId
+              maxCredits
+              minCredits
+              prereqs
+              coreqs
+            }
+          }
+        }`,
+      },
+    });
+
+    const courseData = await res.data.data;
+    if (courseData?.class?.latestOccurrence) {
+      const course: ScheduleCourse = courseData.class.latestOccurrence;
+      course.numCreditsMax = courseData.class.latestOccurrence.maxCredits;
+      course.numCreditsMin = courseData.class.latestOccurrence.minCredits;
+      delete courseData.class.latestOccurrence.maxCredits;
+      delete courseData.class.latestOccurrence.minCredits;
+      return course;
+    } else {
+      return null;
+    }
+  };
+
+  searchCourses = async (
+    searchQuery: string,
+    minIndex = 0,
+    maxIndex = 9999
+  ): Promise<ScheduleCourse[]> => {
+    const res = await this.axios({
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      data: JSON.stringify({
+        query: `
+        {
+          search(termId:"202130", query: "${searchQuery}", classIdRange: {min: ${minIndex}, max: ${maxIndex}}) {
+            totalCount 
+            pageInfo { hasNextPage } 
+            nodes { ... on ClassOccurrence { name subject maxCredits minCredits prereqs coreqs classId
+            } 
+          } 
+        } 
+      }`,
+      }),
+    });
+
+    const coursesData = await res.data;
+    const nodes = coursesData.data.search.nodes;
+
+    const courses = nodes.map((result: SearchClass) => {
+      return {
+        name: result.name,
+        classId: result.classId,
+        subject: result.subject,
+        prereqs: result.prereqs,
+        coreqs: result.coreqs,
+        numCreditsMin: result.minCredits,
+        numCreditsMax: result.maxCredits,
+        semester: null,
+      };
+    });
+
+    return courses;
+  };
+}
+
+export const API = new APIClient(process.env.NEXT_PUBLIC_API_URL);
```

Comment:
let's remove the env var here cause the api is on the same domain as the frontend? and also cause it doesn't work haha